src/Shared/FileUtilities.cs

```diff
@@ -359,41 +359,73 @@ internal static string FixFilePath(string path)
         /// </summary>
         internal static string MaybeAdjustFilePath(string value, string baseDirectory = "")
         {
+            var comparisonType = StringComparison.Ordinal;
+
             // Don't bother with arrays or properties or network paths, or those that
             // have no slashes.
-            if (NativeMethodsShared.IsWindows || string.IsNullOrEmpty(value) ||
-                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\") ||
-                value.IndexOfAny(Slashes) == -1)
+            if (NativeMethodsShared.IsWindows || string.IsNullOrEmpty(value)
+                || value.StartsWith("$(", comparisonType) || value.StartsWith("@(", comparisonType)
+                || value.StartsWith("\\\\", comparisonType))
             {
                 return value;
             }
 
             // For Unix-like systems, we may want to convert backslashes to slashes
-            string newValue = Regex.Replace(value, @"[\\/]+", "/");
+            string newValue = ConvertToUnixSlashes(value);
 
-            string quote = string.Empty;
             // Find the part of the name we want to check, that is remove quotes, if present
-            string checkValue = newValue;
-            if (newValue.Length > 2)
+            bool shouldAdjust = newValue.IndexOf('/') != -1 && LooksLikeUnixFilePath(RemoveQuotes(newValue), baseDirectory);
```

Comment:
I wonder if we could avoid a string allocation by making LooksLikeUnixPath just ignore leading/trailing matching single/double quotes

Reply:
Or give it a `Span<char>` overload, which might help some of its internals too.

Reply:
Is there an alternative to ```ifdef```ing out the ```Span<char>``` overload and related code? I see a ```System.Memory``` for importing ```Span<T>``` outside of netcore 2.1 but I'm guessing because of MSBuildTaskHost I'll have to end up using an ```ifdef``` anyways.

Reply:
Yeah, we will probably start using net46+ `Span` in 16, but blasted .NET 3.5 compat will keep us in `#if` for a long time :(